### PR TITLE
SN-171 & SN-107: Fix bug for storing accesa and video + store images from object detection to S3

### PR DESCRIPTION
**Pull request by hasan-ahmed**
This pull request closes bug request #SN-171 and story #SN-107

**SN-171**

The bug was due to some code changes that made it so the keyClipWriter.start was never execute. This has been fixed as shown in this video:

https://drive.google.com/file/d/1S-26nDAEPsYVc3eyPEa_wh9tQI7VXKYw/view?usp=sharing

**SN-107**

Whenever an object is detected, a jpeg image will be stored in our S3 bucket called images-snapshots. When we train the model next, we can download all of the images on there and add them to our dataset to improve the model. This functionality is shown in this video:

https://drive.google.com/file/d/1ADXBEPHkYheGHx1IqFxc9CNSDCcDs-lG/view?usp=sharing

### CODE REVIEW

**Review by deeppatel-11 — APPROVED** [review]
### _Variables Name._
Variable name 'bucket' can be changed to something more descriptive (line 24)
Variable name 'key' can be changed to something more descriptive (line 24)
 

### _Function Name_
Function name 'put_file' can be 'put_video_file'

### _Long Function._
No Long function in this pull request.

### _Code Complexity._
No code smells, and bugs were detected by the sonarcube. 

### _PR Checklist_

  - [ ] Documentation present for every component and function.
  - [ ] Unit tests are implemented.
  - [x] Implementation matches the corresponding requirements.
  - [x] All commits linked to Jira ticket.
  
### _Overall Remarks._
The problem with storing videos is now solved and the code works on my end.
Approved.

File: WebService/Services/s3_service.py [hunk 1]
```diff
@@ -21,6 +21,14 @@ def upload_file(self, bucket, filepath, key):
             return False
         return True
 
+    def put_file(self, bucket, file_string, key):
```
Comment (MikeG3) [inline WebService/Services/s3_service.py:24]: function name suggestions, for put_file(...):
upload_file(...) or upload_file_to_db(..)

**Review by MikeG3 — COMMENTED** [review]
### _Variables Name._
NA 

### _Function Name_
1 suggestion, WebService/Services/s3_service.py - put_file(...)
upload_file(...) or upload_file_to_db(...)
*Please see inline comment

### _Long Function._
NA

### _Code Complexity._
Sonar Cloud is happy

### _PR Checklist_

  - [ ] Documentation present for every component and function.
  - [ ] Unit tests are implemented.
  - [x] Implementation matches the corresponding requirements.
  - [x] All commits linked to Jira ticket.
  
### _Overall Remarks._
The bug seems to be fixed, fantastic

**Review by BadrSaidi — APPROVED** [review]
### Variables Name
Line 24: the variable name "key" could be more descriptive of the type or functionality of the key.
### Function Name
Line 24: Function name "put_file" could be more descriptive of the functionality of the function.

### Long Function
This pull request contains no long function.

### Code Complexity
No more bugs, and the PR code complexity is very low.

### PR Checklist

- [ ] Documentation present for every component and function.
- [ ] Unit tests are implemented.
- [X] Implementation matches the corresponding requirements.
- [X] All commits linked to Jira ticket.

### Overall Remarks
Storing videos and images from object detection works fine on my end, the bug is fixed.

**Pull request merged**